templates/CRM/common/importProgress.tpl

```diff
@@ -0,0 +1,81 @@
+{*
+ +--------------------------------------------------------------------+
+ | CiviCRM version 4.7                                                |
+ +--------------------------------------------------------------------+
+ | Copyright CiviCRM LLC (c) 2004-2017                                |
+ +--------------------------------------------------------------------+
+ | This file is a part of CiviCRM.                                    |
+ |                                                                    |
+ | CiviCRM is free software; you can copy, modify, and distribute it  |
+ | under the terms of the GNU Affero General Public License           |
+ | Version 3, 19 November 2007 and the CiviCRM Licensing Exception.   |
+ |                                                                    |
+ | CiviCRM is distributed in the hope that it will be useful, but     |
+ | WITHOUT ANY WARRANTY; without even the implied warranty of         |
+ | MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.               |
+ | See the GNU Affero General Public License for more details.        |
+ |                                                                    |
+ | You should have received a copy of the GNU Affero General Public   |
+ | License and the CiviCRM Licensing Exception along                  |
+ | with this program; if not, contact CiviCRM LLC                     |
+ | at info[AT]civicrm[DOT]org. If you have questions about the        |
+ | GNU Affero General Public License or the licensing of CiviCRM,     |
+ | see the CiviCRM license FAQ at http://civicrm.org/licensing        |
+ +--------------------------------------------------------------------+
+*}
+{literal}
+<script type="text/javascript">
+CRM.$(function($) {
+  $("form#Preview").on('submit', function (e) {
+    verify();
+  });
+  function setIntermediate() {
+    var dataUrl = "{/literal}{$statusUrl}{literal}";
+    $.getJSON(dataUrl, function(response) {
+      var dataStr = response.toString();
+      var result  = dataStr.split(",");
+      $("#intermediate").html(result[1]);
+      $("#importProgressBar .ui-progressbar-value").show();
+      if (result[0] < 100) {
+        $("#importProgressBar .ui-progressbar-value").animate({width: result[0] + "%"}, 500);
+        $("#status").text(result[0] + "% Completed");
+      }
+    });
+  }
+
+  function pollLoop() {
+    setIntermediate();
+    window.setTimeout(pollLoop, 10*1000); // 10 sec
```

Comment:
This looks like a potential memory leak as the function is continually calling itself. Also I don't see how this ever ends. It should stop once it reaches 100% but I don't see where that would happen.